src/gs-update-monitor.c

```diff
@@ -788,17 +812,36 @@ gs_update_monitor_init (GsUpdateMonitor *monitor)
 	} else {
 		g_warning ("failed to connect to upower: %s", error->message);
 	}
+
+	network_monitor = g_network_monitor_get_default ();
+	if (network_monitor == NULL)
+		return;
+	monitor->network_monitor = g_object_ref (network_monitor);
+	monitor->network_changed_handler = g_signal_connect (monitor->network_monitor,
+							     "network-changed",
+							     G_CALLBACK (gs_update_monitor_network_changed_cb),
+							     monitor);
 }
 
 static void
 gs_update_monitor_dispose (GObject *object)
 {
 	GsUpdateMonitor *monitor = GS_UPDATE_MONITOR (object);
 
+	if (monitor->network_changed_handler != 0) {
+		g_signal_handler_disconnect (monitor->network_monitor,
+					     monitor->network_changed_handler);
+		monitor->network_changed_handler = 0;
+	}
```

Comment:
I think you should also drop the `GNetworkMonitor` reference that was taken in `init()`.

Reply:
Actually, my comment is plain wrong